### PR TITLE
test: updating jest version to latest

### PR DESCRIPTION
**Pull request by jordanstith15**

### CODE REVIEW

**Review by thomas-c-dillon — APPROVED** [review]
Those look like jest to me

**Review by gavin-everett-genesys — DISMISSED** [review]

File: packages/genesys-spark-components/src/components/stable/gux-calendar/tests/gux-calendar.methods.spec.ts [hunk 1]
```diff
@@ -110,7 +110,7 @@ describe('gux-calendar', () => {
         expect(result).toEqual(dummy);
       });
       it('getRangeDatesElements', () => {
-        spyOn(component, 'getRangeDates').and.callThrough();
+        jest.spyOn(component, 'getRangeDates');
```
Comment (jordanstith15) [inline packages/genesys-spark-components/src/components/stable/gux-calendar/tests/gux-calendar.methods.spec.ts:113]: jest spyOn does "callThrough" by default so this call no longer exists

File: packages/genesys-spark-components/src/components/stable/gux-calendar/tests/gux-calendar.methods.spec.ts [hunk 2]
```diff
@@ -127,10 +127,10 @@ describe('gux-calendar', () => {
         expect(result.length).toEqual(6);
       });
       it('onDateClick with single mode', async () => {
-        spyOn(component, 'setValue').and.callFake(() => {
-          return;
+        jest.spyOn(component, 'setValue').mockImplementation(() => {
+          return Promise.resolve();
         });
-        spyOn(component, 'emitCalendarSelect').and.callFake(() => {
+        jest.spyOn(component, 'emitCalendarSelect').mockImplementation(() => {
```
Comment (jordanstith15) [inline packages/genesys-spark-components/src/components/stable/gux-calendar/tests/gux-calendar.methods.spec.ts:133]: callFake has been removed in favor of mockImplementation when making a custom mock of a function

File: packages/genesys-spark-components/src/components/stable/gux-action-button/tests/gux-action-button.spec.ts [hunk 3]
```diff
@@ -73,7 +73,6 @@ describe('gux-action-button', () => {
   });
 
   it('should fire open and close events if not disabled', async () => {
-    jest.useFakeTimers('legacy');
```
Comment (jordanstith15) [inline packages/genesys-spark-components/src/components/stable/gux-action-button/tests/gux-action-button.spec.ts:76]: This was causing issues with the test exceeding a timeout.

File: packages/genesys-spark-chart-components/package.json [hunk 4]
```diff
@@ -76,11 +76,11 @@
     "glob": "^10.3.0",
     "handlebars": "^4.7.8",
     "inquirer": "^9.2.15",
-    "jest": "^27.5.1",
```
Comment (daragh-king-genesys) [inline packages/genesys-spark-chart-components/package.json:79]: There is a `@types/jest` dependency that should be updated too.

**Review by daragh-king-genesys — COMMENTED** [review]
I get 2 failed snapshot test in the charts package when I run the tests locally

**Review by daragh-king-genesys — APPROVED** [review]
I am still suspicious this was so easy but I can't see anything that is missing or wrong :)

**Pull request merged**